oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/parser/listener/MultiScopesAnalysisListener.java

```diff
@@ -240,12 +240,15 @@ private void setPublicAttrs(SourceBuilder sourceBuilder, SpanObject span) {
         sourceBuilder.setLatency((int) latency);
         sourceBuilder.setResponseCode(Const.NONE);
         span.getTagsList().forEach(tag -> {
-            if (SpanTags.STATUS_CODE.equals(tag.getKey())) {
+            if (SpanTags.HTTP_RESPONSE_STATUS_CODE.equals(tag.getKey())) {
```

Comment:
I think we should check the old tag too? Unless, OAL engine is not compatible with old agent, right? But we should indicate it is `Deprecated` through comment and make the logic clear through comments.

Reply:
Yes, it's reasonable. So I should keep the old tag `SpanTags.STATUS_CODE` and explain in comment that it has been deprecated.

Reply:
I have another question, if application use the new agent, and the tag will be `http.status_code` or `rpc.status_code`, while the old OAL server unrecognized the new tag, so the `responseCode` field will always be 0. Therefore, does agent also need to consider compatibility issues?

Reply:
The agent changes should be applied. And OAP would work with both(deprecated and new)

Reply:
We don't need `new agent`->`old OAP` works. We never recommended that.

Reply:
OK, got it

Reply:
You could add a change log at the project section to highlight this change, and the new agent plugins' response code would not be compatible with old OAP versions.